File: 53. Maximum Subarray.c
int maxSubArray(int* nums, int numsSize){ 
    int maxSum=-10000;    //Limits
    int currentSum=-10000;
    int i;
    for(i=0;i<numsSize;i++){
        currentSum= (nums[i] > currentSum+nums[i] ? nums[i] : currentSum+nums[i]);
        if(maxSum<currentSum)
            maxSum=currentSum;
        
    }
    return maxSum;
}

File: 53. Maximum Subarray.py
from sys import maxsize
class Solution:
    def maxSubArray(self, nums: List[int]) -> int: #Kadane's Algorithm

        maxSum = -maxsize-1
        currentSum=0

        for i in range(len(nums)):
            currentSum += nums[i]

            if currentSum > maxSum:
                maxSum = currentSum

            if currentSum < 0:
                currentSum = 0

            
        return maxSum
